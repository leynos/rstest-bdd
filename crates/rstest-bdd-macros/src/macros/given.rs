//! Implementation of the `#[given]` macro.

use proc_macro::TokenStream;

/// Macro for defining a Given step that registers with the step inventory.
pub(crate) fn given(attr: TokenStream, item: TokenStream) -> TokenStream {
    super::step_attr(attr, item, crate::StepKeyword::Given)
}
