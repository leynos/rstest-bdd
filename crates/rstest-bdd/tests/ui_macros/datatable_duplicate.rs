//! Compile-fail fixture: multiple `#[datatable]` parameters are rejected.

use rstest_bdd_macros::given;

#[given("two datatables")]
fn step(#[datatable] first: Vec<Vec<String>>, #[datatable] second: Vec<Vec<String>>) {}

fn main() {}
