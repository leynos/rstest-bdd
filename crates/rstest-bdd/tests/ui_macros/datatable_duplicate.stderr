error: invalid step function signature: only one DataTable parameter is permitted
 --> tests/ui_macros/datatable_duplicate.rs:6:60
  |
6 | fn step(#[datatable] first: Vec<Vec<String>>, #[datatable] second: Vec<Vec<String>>) {}
  |                                                            ^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: Use `#[given] fn name(...args...)` with supported step arguments/fixtures; remove self.
